content/recipes/interoperability/registering-swisslipids-identifiers.md

```diff
@@ -0,0 +1,162 @@
+needs to be requested. That process is described [here](https://www.wikidata.org/wiki/Wikidata:Property_proposal).
+
+---
+
+
+## Main Content
+
+Making datasets more findable and databases more interoperable, it is recommended to share not just top-level metadata, but also record-level
+metadata. For example, for chemical databases, you can share your chemical compound identifiers as open data, while keeping the data itself
+in your own database. That allows other databases, like Wikidata, to link to your database. First, that makes your data more findable,
+but it also makes your data easier to integrate and therefore more interoperable.
+
+This recipe follows this idea and starts with a CCZero-licensed mapping file, linking InChIKeys and database identifiers.
+Recipe *InChI and SMILES identifiers for chemical structures* ([fcb:FCB007](https://w3id.org/faircookbook/FCB007)) explains how InChIKeys
+can be generated for chemical compounds in your database.
+
+As an example, this recipe demonstrates the approach of listing [SwissLipid](http://www.swisslipids.org/#/) {footcite}`Aimo2015SwissLipids`
+identifiers in Wikidata, developed at the 2021 BioHackathon Europe. Earlier, a [Swiss Lipid property for Wikidata](https://www.wikidata.org/wiki/Wikidata:Property_proposal/SwissLipids_identifier)
+was already proposed, later approved, and created in Wikidata [just before the 2020 BioHackathon Europe](https://www.wikidata.org/w/index.php?title=Property:P8691&oldid=1287579005).
+An existing Wikidata property is a requirement to adding the external identifiers.
+
+The recipe uses [Bacting](https://github.com/egonw/bacting) {footcite}`Willighagen2021` which can be used in Groovy and Python. We
+will use the first here. The starting point is mappings (tuples) that link the SwissLipid identifiers to InChIKeys. The latter
+is used to find the matching Wikidata items.
+
+### Step 1: getting the data
+
+The data is CC-BY but it was agreed that adding the SwissLipid identifiers to Wikidata (CCZero) is okay.
+
+* Download `lipids.tsv` from the [Downloads](https://www.swisslipids.org/#/downloads) page
+* Gunzip the file
+
+### Step 2: extract Swiss Lipid ID <> InChIKey tuples
+
+For this step I use `csvtool` (`apt get install csvtool`):
+
+```shell
+csvtool -t TAB col 1,11 swisslipids.tsv
+```
+
+The output needs some further clean up, like removing lines without InChIKeys or "none" and "-" as value. Also,
+the "InChIKey=" prefix is removed in preparation for the next step. The full used code is:
+
+```shell
+csvtool -t TAB col 1,11 swisslipids.tsv  | sed 's/InChIKey=//' | grep -v "none" | grep -v ",-$" | grep -v ",$" | tee swisslipids_ids.tsv
+```
+
+This results in almost 600k tuples:
+
+```shell
+$ wc -l swiss*tsv
+   592412 swisslipids_ids.tsv
+   777957 swisslipids.tsv
+```
+
+### Step 3: creating a ShEx model
```

Comment:
@egonw:
may be add a statement regarding the selection of shex over a W3C standard (shacl) (even if converters exist).

Reply:
Sorry I missed this, but this step is actually still empty. I will remove it for now.

Reply:
Added SHACL and updated the text a bit.